greenplumpython/table.py

```diff
@@ -1,25 +1,34 @@
 """
-This module creates a Python object :class:`Table` which keeps in memory all the user modifications
-on a table, in order to proceed with SQL query. It concatenates different pieces of queries
-together using CTEs.
+`Table` is the core data structure in GreenplumPython. Conceptually, a `Table`
+is a two-dimensional unordered structure containing data. This aligns with `the
+definition of "Table" on Wikipedia
+<https://en.wikipedia.org/wiki/Table_(information)>`_.
 
-Iterating over rows of a Table can be expensive, so is printing it, or converting it to other data structures
-like :class:`list`, :class:`tuple`, as well as :class:`pandas.DataFrame`. This is because the content need to
-be computed and fetched from a remote database system. That's why :class:`Table` sends the aggregated SQL query
-to the database and returns the final result only when `_fetch()` function is called.
+In the data science world, a `Table` is similar to a `DataFrame` in `pandas
+<https://pandas.pydata.org/>`_, except that
 
-Once the content of a :class:`Table` is fetched from the database system, it will be cached locally for later use.
-Therefore, re-iterating the same table again will be fast.
+- | Data in a `Table` is lazily evaluated rather than eagerly. That is, they are computed only when
+  | they are observed. This can improve efficiency in many cases.
+- | Data in a `Table` is located and manipulated on a remote database system rather than locally. As
+  | a consequence,
 
-Since the content is cached locally, it will become stale once the Table gets modified by someone else on the database system.
-Therefore, you might want to use refresh() to sync the latest update.
-**Note that refresh() is also expensive.**
+    - Retrieving them from the database system can be expensive, and
+    - They might be modified concurrently by other users of the database system.
 
+  | Due to the first consequence, once the data of a :class:`Table` is fetched from
+  | the database system, it will be cached locally for later use.
```

Comment:
> it will be cached locally for later use.

I don't quite get this, what does "cached locally" mean here? Is this the iterator case we discussed before?

Reply:
hmm, OK. We may want to change the statement when we support cursor. caching data locally from MPP DB may sound scary to users.